crates/data_components/src/databricks/sql_warehouse.rs

```diff
@@ -0,0 +1,655 @@
+        "TIMESTAMP" => DataType::Timestamp(TimeUnit::Microsecond, Some("UTC".into())),
+        "TIMESTAMP_NTZ" => DataType::Timestamp(TimeUnit::Microsecond, None),
+        "VOID" => DataType::Null,
+        ty => return Err(Error::UnsupportedType { ty: ty.to_string() }),
+    })
+}
+
+fn schema_from_json(json_value: &Value) -> Result<SchemaRef, Error> {
+    let data_array = json_value
+        .get("result")
+        .and_then(|r| r.get("data_array"))
+        .and_then(|d| d.as_array())
+        .ok_or_else(|| Error::UnableToRetrieveSchema {
+            reason: "result.data_array".to_string(),
+        })?;
+
+    let mut fields = Vec::new();
+
+    for (i, row) in data_array.iter().enumerate() {
+        let row_array = row
+            .as_array()
+            .ok_or_else(|| Error::UnableToRetrieveSchema {
+                reason: format!("data_array[{i}] is not an array"),
+            })?;
+
+        if row_array.len() < 2 {
+            return Err(Error::UnableToRetrieveSchema {
+                reason: format!("data_array[{i}] lacks col_name or data_type"),
+            });
+        }
+
+        let col_name = row_array[0]
+            .as_str()
+            .ok_or_else(|| Error::UnableToRetrieveSchema {
+                reason: format!("data_array[{i}][0] is not a string"),
+            })?;
+
+        // If we see #, DB is now providing the clustering metadata info and
+        // we have all the columns we need
+        if col_name.starts_with('#') {
+            break;
+        }
+
+        let data_type_str = row_array[1]
+            .as_str()
+            .ok_or_else(|| Error::UnableToRetrieveSchema {
+                reason: format!("data_array[{i}][1] is not a string"),
+            })?;
+
+        let field = Field::new(col_name, map_databricks_type(data_type_str)?, true);
+
+        fields.push(field);
+    }
+
+    Ok(Arc::new(Schema::new(fields)))
+}
+
+struct SqlWarehouseConnection {
+    api: Arc<SqlWarehouseApi>,
+}
+
+impl<'a> DbConnection<Arc<SqlWarehouseApi>, &'a dyn Sync> for SqlWarehouseConnection {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+
+    fn as_async(&self) -> Option<&dyn AsyncDbConnection<Arc<SqlWarehouseApi>, &'a dyn Sync>> {
+        Some(self)
+    }
+}
+
+#[async_trait]
+impl<'a> AsyncDbConnection<Arc<SqlWarehouseApi>, &'a dyn Sync> for SqlWarehouseConnection {
+    fn new(api: Arc<SqlWarehouseApi>) -> Self {
+        Self { api }
+    }
+
+    async fn get_schema(
+        &self,
+        table_reference: &TableReference,
+    ) -> Result<SchemaRef, dbconnection::Error> {
+        self.api
+            .get_schema(table_reference)
+            .await
+            .map_err(|source| dbconnection::Error::UnableToGetSchema {
+                source: Box::new(source),
+            })
+    }
+
+    async fn query_arrow(
+        &self,
+        sql: &str,
+        _: &[&'a dyn Sync],
```

Comment:
Does the Databricks SQL warehouse API support parameterized queries?

Reply:
yes it does! I don't fully understand yet how we would use it here yet. It would be trivial if `AsyncDbConnection::query_arrow` had arguments for it here.

Reply:
I didn't realize that the `AsyncDbConnection::execute` takes parameters. I'll have to look into this more.

Reply:
Yes it does! I can see how we can add that in a follow-up.